zipline/assets/asset_writer.py

```diff
@@ -563,3 +560,9 @@ def _load_data(self, equities, futures, exchanges, root_symbols):
             exchanges=exchanges_output,
             root_symbols=root_symbols_output,
         )
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *exc_info):
+        pass
```

Comment:
The only thing that the cm is capable of closing is the SqlAlchemy engine, which has a dispose method. It closes all the checked-in connections. All of the checked out connections will be left open, however. 

I'm not exactly sure if this method should try to close all connections, or just those that are not in use